sonar-slang-plugin/src/test/java/com/sonarsource/slang/externalreport/androidlint/AndroidLintSensorTest.java

```diff
@@ -0,0 +1,178 @@
+
+  @Test
+  public void issues_with_sonarqube_72() throws IOException {
+    SensorContextTester context = createContext(PROJECT_DIR, 7, 2);
+    context.settings().setProperty("sonar.androidLint.reportPaths", resolveInProject("lint-results.xml"));
+    List<ExternalIssue> externalIssues = executeSensor(detektSensor, context);
+    assertThat(externalIssues).hasSize(4);
+
+    ExternalIssue first = externalIssues.get(0);
+    assertThat(first.primaryLocation().inputComponent().key()).isEqualTo("androidlint-project:AndroidManifest.xml");
+    assertThat(first.ruleKey().toString()).isEqualTo("android-lint-xml:AllowBackup");
+    assertThat(first.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(first.severity()).isEqualTo(Severity.MINOR);
+    assertThat(first.primaryLocation().message()).isEqualTo(
+      "On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html");
+    assertThat(first.primaryLocation().textRange().start().line()).isEqualTo(2);
+
+    ExternalIssue second = externalIssues.get(1);
+    assertThat(second.primaryLocation().inputComponent().key()).isEqualTo("androidlint-project:A.java");
+    assertThat(second.ruleKey().toString()).isEqualTo("android-lint-java:GoogleAppIndexingWarning");
+    assertThat(second.primaryLocation().textRange().start().line()).isEqualTo(1);
+
+    ExternalIssue third = externalIssues.get(2);
+    assertThat(third.primaryLocation().inputComponent().key()).isEqualTo("androidlint-project:B.kt");
+    assertThat(third.ruleKey().toString()).isEqualTo("android-lint-kotlin:GoogleAppIndexingWarning");
+    assertThat(third.primaryLocation().textRange().start().line()).isEqualTo(2);
+
+    ExternalIssue fourth = externalIssues.get(3);
+    assertThat(fourth.primaryLocation().inputComponent().key()).isEqualTo("androidlint-project:build.gradle");
+    assertThat(fourth.ruleKey().toString()).isEqualTo("android-lint:GradleDependency");
+    assertThat(fourth.primaryLocation().textRange().start().line()).isEqualTo(3);
+
+    assertLogsContainOnlyInfo(logTester);
+  }
+
+  @Test
+  public void no_issues_without_report_paths_property() throws IOException {
+    SensorContextTester context = createContext(PROJECT_DIR, 7, 2);
+    List<ExternalIssue> externalIssues = executeSensor(detektSensor, context);
+    assertThat(externalIssues).isEmpty();
+    assertLogsContainOnlyInfo(logTester);
+  }
+
+  @Test
+  public void no_issues_with_invalid_report_path() throws IOException {
+    SensorContextTester context = createContext(PROJECT_DIR, 7, 2);
+    context.settings().setProperty("sonar.androidLint.reportPaths", resolveInProject("invalid-path.txt"));
+    List<ExternalIssue> externalIssues = executeSensor(detektSensor, context);
+    assertThat(externalIssues).isEmpty();
```

Comment:
these 3 assertions could be refactored

Reply:
refactored into "executeSensorImporting("lint-results.xml")"